### PR TITLE
Side-by-side current and legacy version of uap-core

### PR DESCRIPTION
**Pull request by strato-cumulus**
This is a follow-up to the question whether we can use two different versions of the UA parser in DMP API. We can't import two versions of the same library directly, but since we are using a custom release anyway, I decided to modify it a bit more.
This release adds factory methods to `Parser` and `CachedParser` which yield a parser built on either current `uap-core` version or a legacy one (the version the current master branch uses). The DMP API will then obtain two separate parser instances whose output will be consumed in two different models.
This PR would bump the version to `1.5.2-mbr`, reflecting the parent repo version we rebased against.

### CODE REVIEW

File: src/test/java/ua_parser/CachingParserTest.java [hunk 1]
```diff
@@ -84,4 +92,19 @@ public void testCachedReplacementQuoting() throws Exception {
...
+    Client expected1 = new Client(new UserAgent("Firefox", "3", "5", "5"),
+                                  new OS("Mac OS X", "10", "4", null, null),
+                                  new Device("Mac", "Apple", "Mac"));
+    Client expected2 = new Client(new UserAgent("Mobile Safari", "5", "1", null),
+                                  new OS("iOS", "5", "1", "1", null),
+                                  new Device("iPhone", "Apple", "iPhone"));
+
+    assertThat(parser.parse(agentString1), is(expected1));
+    assertThat(parser.parse(agentString2), is(expected2));
+  }
```
Comment (einhorntee) [inline src/test/java/ua_parser/CachingParserTest.java:108]: [these](https://github.com/mbrtargeting/popper/blob/master/popper/segmem/data_parser.py#L224) are the features used from the python side - are we sure that this is going to translate 100% to the same set of features?

File: src/test/java/ua_parser/CurrentParserTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        String agentString1 = "Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.4; fr; rv:1.9.1.5) Gecko/20091102 Firefox/3.5.5,gzip(gfe),gzip(gfe)";
+        String agentString2 = "Mozilla/5.0 (iPhone; CPU iPhone OS 5_1_1 like Mac OS X) AppleWebKit/534.46 (KHTML, like Gecko) Version/5.1 Mobile/9B206 Safari/7534.48.3";
+
+        Client expected1 = new Client(new UserAgent("Firefox", "3", "5", "5"),
+                                      new OS("Mac OS X", "10", "4", null, null),
+                                      new Device("Mac", "Apple", "Mac"));
+        Client expected2 = new Client(new UserAgent("Mobile Safari", "5", "1", null),
+                                      new OS("iOS", "5", "1", "1", null),
+                                      new Device("iPhone", "Apple", "iPhone"));
+
```
Comment (einhorntee) [inline src/test/java/ua_parser/CurrentParserTest.java:31]: same as above

File: src/test/java/ua_parser/LegacyParserTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        String agentString1 = "Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.4; fr; rv:1.9.1.5) Gecko/20091102 Firefox/3.5.5,gzip(gfe),gzip(gfe)";
+        String agentString2 = "Mozilla/5.0 (iPhone; CPU iPhone OS 5_1_1 like Mac OS X) AppleWebKit/534.46 (KHTML, like Gecko) Version/5.1 Mobile/9B206 Safari/7534.48.3";
+
+        Client expected1 = new Client(new UserAgent("Firefox", "3", "5", "5"),
+                                      new OS("Mac OS X", "10", "4", null, null),
+                                      new Device("Other", null, null));
+        Client expected2 = new Client(new UserAgent("Mobile Safari", "5", "1", null),
+                                      new OS("iOS", "5", "1", "1", null),
+                                      new Device("iPhone", "Apple", "iPhone"));
+
```
Comment (einhorntee) [inline src/test/java/ua_parser/LegacyParserTest.java:31]: hang on, why are there 3 tests for 2 parser versions?

**Review by einhorntee — COMMENTED** [review]
unfortunately, I can't comment on files that changed, but the `uap-core-legacy` points to ` fc570f` of the uap-core, but the [outdated python version in popper ](https://github.com/mbrtargeting/popper/blob/master/requirements.txt#L64) is linked to [0.7.3](https://github.com/ua-parser/uap-python/tree/0.7.3) which is pointing to `uap-core @ ce89c76`.
The [newest available python version 0.10.0](https://github.com/ua-parser/uap-python/tree/0.10.0) is based on `uap-core @ ec8322f`, so that'd need to be the version the _new_ parser points to.

File: src/test/java/ua_parser/CachingParserTest.java (hunk 1 above)
Comment (strato-cumulus) [reply src/test/java/ua_parser/CachingParserTest.java:108]: I only moved that test :smile: It has been updated for the new model though, the more recent revision of `uap-core` can pick up that it's a Mac. As you can see in the `LegacyParserTest`, it has been detected as `Other` previously.

File: src/test/java/ua_parser/LegacyParserTest.java (hunk 3 above)
Comment (strato-cumulus) [reply src/test/java/ua_parser/LegacyParserTest.java:31]: Honestly a `CachingParserTest` uses the `Current` model... I will make this a special case of `CurrentParserTest`.

**Pull request closed without merging**